packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+    // one, so we have to account for it here)
+    const allTenantIds = getAllTenantIds(dbContents);
+    const agencyOptionsByTenantId = _.fromPairs(
+        allTenantIds.map((tenantId) => [
+            tenantId,
+            dbContents.agencies.filter(
+                (agency) => agency.tenant_id === tenantId
+            ),
+        ])
+    );
+    const questions = _.chain(allTenantIds)
+        .map((tenantId) => [
+            {
+                type: "input",
+                name: `tenantNames.${tenantId}`,
+                message: `Tenant name for source tenant ${tenantId}`,
+                validate: (name) =>
+                    name.trim().length > 0 || "must enter a name",
+            },
+            {
+                type: "search-list",
+                name: `mainAgencies.${tenantId}`,
+                message: `Main agency for source tenant ${tenantId}`,
+                choices: [
+                    {
+                        name: "(create new agency with same name as tenant)",
+                        value: null,
+                    },
+                    ...agencyOptionsByTenantId[tenantId].map((agency) => ({
+                        name: agency.name,
+                        value: agency.id,
+                    })),
+                ],
+            },
+        ])
+        .flatten()
+        .value();
+    let { tenantNames, mainAgencies } = await inquirer.prompt(questions);
+
+    // Annoying: by default Inquirer will make (sparse) arrays instead of objects; convert them back
+    // to the expected form.
+    tenantNames = _.fromPairs(
+        tenantNames
+            .filter((x) => x !== undefined)
+            .map((name, idx) => [idx, name])
+    );
+    mainAgencies = _.fromPairs(
+        mainAgencies
+            .filter((x) => x !== undefined)
+            .map((agencyId, idx) => [idx, agencyId])
+    );
+
+    // Create tenants
+    let inserted = [];
+    for (const tenantId of allTenantIds) {
+        const tenantName = tenantNames[tenantId];
+        const mainAgencyId = mainAgencies[tenantId];
+
+        const tenant = await knex("tenants")
+            .insert({
+                display_name: tenantName,
+                // main_agency_id must be populated in a second pass in importAgencies
+                // because there is a circular FK between tenants/agencies tables
+                main_agency_id: null,
+            })
+            .returning("*");
+        inserted.push(tenant);
+
+        if (mainAgencyId) {
+            tenant.__unmappedFutureMainAgencyId = mainAgencyId;
+        } else {
+            const agencyToInsert = {
+                id: 1000000 + tenantId,
+                tenant_id: tenantId,
+                name: tenantName,
+                code: `CHANGEME_${tenant.id}`,
+                __isMainAgency: true,
+            };
+            dbContents.agencies.splice(0, 0, agencyToInsert);
```

Comment:
Can we print an output statement here mentioning that we're adding in a new agency, so that it's clear why the import counts don't match the export counts?

I guess it isn't super critical because the inquirer question lets them know we are creating a new agency, but might be less confusing to be explicit when we actually add it in the code

Reply:
Sure. In practice, the existing deployments have a single tenant (ID `0`) so there would be at most a single new agency created, so I don't think it's too confusing, but adding logging is easy.